pkg/controller/baremetalasset/baremetalasset_controller.go

```diff
@@ -295,13 +305,93 @@ func (r *ReconcileBareMetalAsset) Reconcile(request reconcile.Request) (reconcil
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileBareMetalAsset) checkHiveSyncSetInstance(bma *midasv1alpha1.BareMetalAsset, reqLogger logr.Logger) error {
+	found := &hivev1.SyncSetInstanceList{}
+	err := r.client.List(context.TODO(), found, client.MatchingLabels{hiveconstants.SyncSetNameLabel: bma.Name})
+
+	if err != nil {
+		reqLogger.Error(err, "Failed to get Hive SyncSetInstance")
+		conditionsv1.SetStatusCondition(&bma.Status.Conditions, conditionsv1.Condition{
+			Type:    midasv1alpha1.ConditionAssetSyncCompleted,
+			Status:  corev1.ConditionFalse,
+			Reason:  "SyncSetInstanceRetreivalFailed",
+			Message: "Failed to get SyncSetInstance",
+		})
+		return err
+	}
+	switch len(found.Items) {
+	case 0:
+		conditionsv1.SetStatusCondition(&bma.Status.Conditions, conditionsv1.Condition{
+			Type:    midasv1alpha1.ConditionAssetSyncCompleted,
+			Status:  corev1.ConditionFalse,
+			Reason:  "SyncSetInstanceNotFound",
+			Message: "SyncSetInstance not found",
+		})
+		return errors.New("Failed to get Hive SyncSetInstance")
+	case 1:
+		syncSetInstanceResource := found.Items[0].Status.Resources[0]
+		if syncSetInstanceResource.Kind == BareMetalHostKind && syncSetInstanceResource.APIVersion == BareMetalHostAPIVersion {
+			for _, condition := range syncSetInstanceResource.Conditions {
+				switch condition.Type {
+				case hivev1.ApplySuccessSyncCondition:
+					conditionsv1.SetStatusCondition(&bma.Status.Conditions, conditionsv1.Condition{
+						Type:    midasv1alpha1.ConditionAssetSyncCompleted,
+						Status:  condition.Status,
+						Reason:  condition.Reason,
+						Message: condition.Message,
+					})
+				case hivev1.ApplyFailureSyncCondition:
+					conditionsv1.SetStatusCondition(&bma.Status.Conditions, conditionsv1.Condition{
+						Type:    midasv1alpha1.ConditionAssetSyncCompleted,
+						Status:  corev1.ConditionFalse,
+						Reason:  condition.Reason,
+						Message: condition.Message,
+					})
+					return errors.New("Problem propagating a BMA to a managed host")
+				}
+			}
+		}
+		syncSetInstanceSecret := found.Items[0].Status.Secrets[0]
+		if syncSetInstanceSecret.Kind == SecretKind && syncSetInstanceSecret.APIVersion == SecretAPIVersion {
+			for _, condition := range syncSetInstanceSecret.Conditions {
+				switch condition.Type {
+				case hivev1.ApplySuccessSyncCondition:
+					conditionsv1.SetStatusCondition(&bma.Status.Conditions, conditionsv1.Condition{
+						Type:    midasv1alpha1.ConditionAssetSyncCompleted,
+						Status:  condition.Status,
+						Reason:  condition.Reason,
+						Message: condition.Message,
+					})
+				case hivev1.ApplyFailureSyncCondition:
+					conditionsv1.SetStatusCondition(&bma.Status.Conditions, conditionsv1.Condition{
+						Type:    midasv1alpha1.ConditionAssetSyncCompleted,
+						Status:  corev1.ConditionFalse,
+						Reason:  condition.Reason,
+						Message: condition.Message,
+					})
+					return errors.New("Problem propagating a BMA to a managed host")
```

Comment:
You should call out the fact that we are having a problem with the secret here.
```suggestion
					return errors.New("Problem propagating BareMetalHost credentials to managed cluster")
```

Reply:
Done